### DISCUSSION TITLE
How do I get location of a certain block in PDF?

### DISCUSSION THREAD

**Discussion by shubhamdrip** [prawnpdf/prawn#1275] [Q&A] [answered]
I want to add an image to a PDF. The PDF contains a rectangular box, whose position I do not know beforehand, inside which I need to add the image. How do I find the location of the box?

**Comment by pointlessone** [comment, marked as answer]
Prawn doesn't provide anything for that. Prawn is concerned with generation of PDF and doesn't have any API to inspect the content.

That said, there's `pdf-inspector` gem that does specifically that. PDF is not very good at providing metadata to identify a specific rectangle but maybe you'll have some luck with the gem.
